### DISCUSSION TITLE
How to apply goimports

### DISCUSSION THREAD

**Discussion by juniway** [helix-editor/helix#2175] [Q&A]
Does the LSP support for golang has `goimports` enabled? How can I format the code with goimports?

**Comment by sudormrfbin** [comment]
See https://github.com/helix-editor/helix/issues/2139.

**Reply by juniway** [reply]
thank you, but actually I tried with `:pipe goimports`, it only prompts me error with this:
<img width="409" alt="CleanShot 2022-04-19 at 23 41 40@2x" src="https://user-images.githubusercontent.com/3980841/164042816-5c941665-5c71-491e-8616-b8052180975b.png">

**Reply by the-mikedavis** [reply]
The pipe typable command was merged recently (after 22.03) so you'll need to build from source to be able to use `:pipe`.

**Reply by juniway** [reply]
I pull the latest code, rebuilt **`hx`**, and now it's `helix 22.05-dev (cc68fa85)`, I tested again with `:pipe goimports`, still, nothing happened.

**Reply by the-mikedavis** [reply]
Pipe (also `|` in normal mode) takes a selection as input, so if you'd like to format the entire buffer you'll need to first select the buffer with `%` and then pipe through goimports.

**Comment by juniway** [comment]
Now it works, thanks a lot. Could it be possible to make it executed
automatically on save? Kinda like a autocmd in vim
